src/main/java/org/terasology/combatSystem/weaponFeatures/events/PrimaryAttackEvent.java

```diff
@@ -27,6 +28,14 @@ public PrimaryAttackEvent(ActivateEvent info){
         hitPosition = info.getHitPosition();
         hitNormal = info.getHitNormal();
         activationId = info.getActivationId();
+        EntityRef entity1 = getInstigator();
+        EntityRef entity2 = getTarget();
+        LocationComponent locI = entity1.getComponent(LocationComponent.class);
+        LocationComponent locT = entity2.getComponent(LocationComponent.class);
+        Vector3f impulse = new Vector3f(locT.getWorldPosition()).sub(locI.getWorldPosition());
+        impulse.normalize();
+        impulse.scale(5);
+        entity2.send(new CharacterImpulseEvent(impulse));
```

Comment:
This only works if the target (`entity`) is defined, it will throw a `NullPointerException` otherwise. You can easily observe that by hitting into plain air with your sword, or firing arrows.

You can guard against this by checking whether the entity is defined:
```suggestion
		if (instigator.exists() && target.exists()) {
	        LocationComponent locI = entity1.getComponent(LocationComponent.class);
	        LocationComponent locT = entity2.getComponent(LocationComponent.class);
	        Vector3f impulse = new Vector3f(locT.getWorldPosition()).sub(locI.getWorldPosition());
	        impulse.normalize();
	        impulse.scale(5);
	        entity2.send(new CharacterImpulseEvent(impulse));
        }
```

Reply:
This is how it looks like in the logs:
```
16:04:22.680 [main] ERROR o.t.e.event.internal.EventSystemImpl - Failed to invoke event
java.lang.NullPointerException: null
        at org.terasology.combatSystem.weaponFeatures.events.PrimaryAttackEvent.<init>(PrimaryAttackEvent.java:35)
        at org.terasology.combatSystem.weaponFeatures.systems.AttackSystem.primaryAttack(AttackSystem.java:22)
        at org.terasology.combatSystem.weaponFeatures.systems.AttackSystemMethodAccess.invoke(Unknown Source)
        at org.terasology.entitySystem.event.internal.EventSystemImpl$ByteCodeEventHandlerInfo.invoke(EventSystemImpl.java:531)
        at org.terasology.entitySystem.event.internal.EventSystemImpl.sendConsumableEvent(EventSystemImpl.java:301)
        at org.terasology.entitySystem.event.internal.EventSystemImpl.send(EventSystemImpl.java:280)
        at org.terasology.entitySystem.entity.internal.BaseEntityRef.send(BaseEntityRef.java:205)
        at org.terasology.logic.characters.CharacterSystem.onActivationRequest(CharacterSystem.java:282)
        at org.terasology.logic.characters.CharacterSystemMethodAccess.invoke(Unknown Source)
        at org.terasology.entitySystem.event.internal.EventSystemImpl$ByteCodeEventHandlerInfo.invoke(EventSystemImpl.java:531)
        at org.terasology.entitySystem.event.internal.EventSystemImpl.sendStandardEvent(EventSystemImpl.java:291)
        at org.terasology.entitySystem.event.internal.EventSystemImpl.send(EventSystemImpl.java:282)
        at org.terasology.entitySystem.entity.internal.BaseEntityRef.send(BaseEntityRef.java:205)
        at org.terasology.logic.players.LocalPlayer.activateTargetOrOwnedEntity(LocalPlayer.java:231)
        at org.terasology.logic.players.LocalPlayer.activateOwnedEntityAsClient(LocalPlayer.java:185)
        at org.terasology.logic.players.LocalPlayerSystem.onUseItemButton(LocalPlayerSystem.java:485)
        at org.terasology.logic.players.LocalPlayerSystemMethodAccess.invoke(Unknown Source)
        at org.terasology.entitySystem.event.internal.EventSystemImpl$ByteCodeEventHandlerInfo.invoke(EventSystemImpl.java:531)
        at org.terasology.entitySystem.event.internal.EventSystemImpl.sendConsumableEvent(EventSystemImpl.java:301)
        at org.terasology.entitySystem.event.internal.EventSystemImpl.send(EventSystemImpl.java:280)
        at org.terasology.entitySystem.entity.internal.BaseEntityRef.send(BaseEntityRef.java:205)
        at org.terasology.input.internal.BindableButtonImpl.updateBindState(BindableButtonImpl.java:163)
        at org.terasology.input.InputSystem.updateBindState(InputSystem.java:317)
        at org.terasology.input.InputSystem.processMouseButtonInput(InputSystem.java:222)
        at org.terasology.input.InputSystem.processMouseInput(InputSystem.java:198)
        at org.terasology.input.InputSystem.update(InputSystem.java:134)
        at org.terasology.engine.modes.StateIngame.handleInput(StateIngame.java:192)
        at org.terasology.engine.subsystem.lwjgl.LwjglInput.postUpdate(LwjglInput.java:55)
        at org.terasology.engine.TerasologyEngine.tick(TerasologyEngine.java:471)
        at org.terasology.engine.TerasologyEngine.mainLoop(TerasologyEngine.java:425)
        at org.terasology.engine.TerasologyEngine.run(TerasologyEngine.java:401)
        at org.terasology.engine.Terasology.main(Terasology.java:162)
```